cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -82,18 +82,25 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 
 	// Get the cluster version
 	clusterVersionStr, clusterVersion, err := versionGetterImpl.ClusterVersion()
+	if len(clusterVersionStr) > 0 {
```

Comment:
In general, if `err != nil` we should not look at any other return values of funcs. Also, if there are no errors, these strings should be non-empty and printable (hence no checks are required).
Can you do the prints as else clauses to the error checks?
The same applies practically everywhere throughout this patch.

Reply:
Fixed.